File: src/main/java/com/sothawo/springdataelastictest/person/PersonRepository.java
/*
 * (c) Copyright 2021 sothawo
 */
package com.sothawo.springdataelastictest.person;

import com.sothawo.springdataelastictest.RoutingAwareReactiveElasticsearchRepository;
import org.springframework.data.elasticsearch.core.SearchHit;
import reactor.core.publisher.Flux;

/**
 * @author P.J. Meisch (pj.meisch@sothawo.com)
 */
public interface PersonRepository extends RoutingAwareReactiveElasticsearchRepository<Person, Long> {
		Flux<SearchHit<Person>> searchByLastName(String lastName);
}

File: src/main/kotlin/com/sothawo/springdataelastictest/person/PersonRepository.kt
package com.sothawo.springdataelastictest.person

import kotlinx.coroutines.flow.Flow
import org.springframework.data.elasticsearch.core.SearchHit
import org.springframework.data.elasticsearch.repository.CoroutineElasticsearchRepository
import reactor.core.publisher.Flux

/**
 * @author P.J. Meisch (pj.meisch@sothawo.com)
 */
interface PersonRepository : CoroutineElasticsearchRepository<Person, Long> {

		// one function to return a flow
		suspend fun findByLastName(lastName: String): Flow<SearchHit<Person>>
}
